hbase-client/src/main/java/org/apache/hadoop/hbase/client/ConnectionFactory.java

```diff
@@ -225,7 +256,7 @@ public static Connection createConnection(Configuration conf, ExecutorService po
           clazz.getDeclaredConstructor(Configuration.class, ExecutorService.class, User.class);
         constructor.setAccessible(true);
         return user.runAs((PrivilegedExceptionAction<
-          Connection>) () -> (Connection) constructor.newInstance(conf, pool, user));
+          Connection>) () -> (Connection) constructor.newInstance(conf, pool, user, identity));
```

Comment:
the metrics don't seem to be tracked via this constructor. They are in the `AsyncConnection` path, and this is for `Connection`.

Reply:
Your are right, there should no need to change here. Updated.